recipes/spdlog/all/conanfile.py

```diff
@@ -42,8 +42,6 @@ def config_options(self):
     def configure(self):
         if self.options.shared or self.options.header_only:
             self.options.rm_safe("fPIC")
-        if self.options.header_only:
-            del self.options.shared
```

Comment:
why?

Reply:
One word reply’s are generally seen as rude so I love to encourage everyone to use short bullet point style sentences, make its much more agreeable to answer helpfully :)

Everything is removed by the `self.info.clear()` in the package ID (which was not so with the old `header_only` way) so this does not have an impact or duplicated what the package id's effect is. I opted to remove it here just to have a current best example

Reply:
It exposes shared as a possible option for consumers while it doesn't make sense for them. Moreover it breaks hooks.

Reply:
We can fix the hooks, and it only ignore the imput from users in my experience so it's the same UX.

You can put both and it's acceptable